### DISCUSSION TITLE
What's the purpose of `emailVisibility`?

### DISCUSSION THREAD

**Discussion by teenjuna** [pocketbase/pocketbase#1442] [Q&A] [answered]
The question is pretty much in title. I couldn't find any information about this field. What it really does?

**Comment by ganigeorgiev** [comment, marked as answer]
You can find information for the available auth record fields in the "Admin UI > (your auth collection) > API Preview > Create or Update" docs (alternative you can also find this info in the general Records Create/Update docs - https://pocketbase.io/docs/api-records/).

In short, it makes the user email address public or private (the user owner, admins and those that satisfies the Manage API rule of the collection can still see the email). This is useful for example if you want to allow a public list with your users.

**Reply by teenjuna** [reply]
Thanks for such a quick answer! I understand the example, but why is it built-in in the system? If I want to have such a niche thing like having a public list of users and allowing users to hide themselves from the list, I can just add this field myself in the Admin UI. Isn't that so? Especially because I as developer can just reset the field through hooks...

Actually, I don't even understand why we have both `username` and `email` build-in. It just looks very opinionated (not to insult the project, I'm fascinated by it and your hard work!). The email is special, because you support email-related stuff (verification, etc), but there could be an option to disable this field completely if user doesn't want it. For example, I'm trying to build an e-commerce site, where by default an anonymous user is created (password is generated on client, username is generated by PocketBase). Then, there's an option to "upgrade" the user by authenticating through OAuth provider (making the user `registered`). I don't really need `email` at all. I know that I can make it not required, but it would be cool if I could disable it completely to make my system more clear.

To summarize, current user management looks a bit opinionated for some use cases. I fully understand that it could be a consequence of previous design decisions and/or vision of the project. But is it a possibility that you will consider making it more generic with opt-in things like email?

I want to point out again this is not a "feature demand with no respect for OS maintainers" but an honest question.

**Reply by ganigeorgiev** [reply]
>  If I want to have such a niche thing like having a public list of users and allowing users to hide themselves from the list, I can just add this field myself in the Admin UI. Isn't that so? 

Unfortunately it is not that simple. You'll have to [extend PocketBase](https://pocketbase.io/docs/use-as-framework/) if you want to emulate the `emailVisibility` behavior. It is not just a "show/hide" toggle. Even if you add the field on your own you'll have to bind all the necessary custom requirements who can see, filter, sort the email address.

> Actually, I don't even understand why we have both username and email build-in.

It doesn't hurt having the fields by default and it helps reducing the verbosity and boilerplate. Additionally, as you noted, the email is always available because it is used in various auth related scenarios (email verification, password reset, OAuth2 account linking, etc.). We can't just remove it since it will add fair amount of complexity and confusion without too much benefit.

>  But is it a possibility that you will consider making it more generic with opt-in things like email?

No, sorry there are no plans for that at the moment. PocketBase will be always "restricted" to someone and may not satisfy all use cases. And that's OK. It is definitely not a "one-size-fit-all" type of project. I've built it mainly because I need it for my other open source project - [Presentator #183](https://github.com/presentator/presentator/issues/183).

_If your app require more "flexibility" there are already plenty of other well made open source BaaS-like platforms that may be more suitable for your use case - [Supabase](https://supabase.com/), [Nhost](https://nhost.io/), [Appwrite](https://appwrite.io/), [EdgeDB](https://www.edgedb.com/), [SurrealDB](https://surrealdb.com/) and probably others. My advice would be to give them a try and see what best fit with your use case and desired workflow._

**Reply by teenjuna** [reply]
Thanks for the answer! Now I understand your point :)
